platform/security/certificate/src/main/java/org/codice/ddf/certificate/CertificateManager.java

```diff
@@ -0,0 +1,267 @@
+     * <p/>
+     * Purpose:  Registers BouncyCastle as a java.Security provider
+     * <p/>
+     * Pre-condition:  None
+     * <p/>
+     * Post-condition: Bouncy Castle is added to the list of java.Security Providers
+     * <p/>
+     * Parameters:     None
+     * <p/>
+     * Returns:        None
+     * -------------------------------------------------------------------
+     */
+
+    public static void registerBCSecurityProvider() {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    /**
+     * ---------------------------------------------------------------------
+     * Method: getHostName
+     * <p/>
+     * Purpose:  Used to retrieve the hostname of the machine
+     * <p/>
+     * Pre-condition:  None
+     * <p/>
+     * Post-condition: Hostname is returned
+     * <p/>
+     * Parameters:     None
+     *
+     * @return String representing the hostname
+     * -------------------------------------------------------------------
+     */
+
+    public static String getHostName() {
+        String hostName = null;
+        try {
+            hostName = InetAddress.getLocalHost().getHostName();
+        } catch (UnknownHostException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        return hostName;
+    }
+
+
+    /*+----------------------------------------------------------------------
+    ||
+    ||  Class: SHA256RSASigner
+    ||
+    ||        Purpose:  Implementation of ContentSigner that uses SHa256withRSA. Used to sign X509 certificates
+    ||
+    ||     Interfaces:  ContentSigner
+    ||
+    ++-----------------------------------------------------------------------*/
+
+    private static class SHA256RSASigner implements ContentSigner {
+
+        private static final AlgorithmIdentifier SHA_256_WITH_RSA_ID = new AlgorithmIdentifier(
+                new ASN1ObjectIdentifier("1.2.840.113549.1.1.11"));
+
+        private Signature signature;
+
+        private ByteArrayOutputStream outputStream;
+
+        public SHA256RSASigner(PrivateKey privateKey) {
+            try {
+                this.outputStream = new ByteArrayOutputStream();
+                this.signature = Signature.getInstance("SHA256withRSA");
+                this.signature.initSign(privateKey);
+            } catch (GeneralSecurityException e) {
+                LOGGER.error(e.getMessage(), e);
+            }
+        }
+
+        @Override
+        public AlgorithmIdentifier getAlgorithmIdentifier() {
+            return SHA_256_WITH_RSA_ID;
+        }
+
+        @Override
+        public OutputStream getOutputStream() {
+            return outputStream;
+        }
+
+        @Override
+        public byte[] getSignature() {
+            try {
+                signature.update(outputStream.toByteArray());
+                return signature.sign();
+            } catch (GeneralSecurityException e) {
+                LOGGER.error(e.getMessage(), e);
+                return null;
```

Comment:
If we keep this method and make it public, make sure we clearly define what it does and what it can return. In other words, should it return null or throw an exception?